src/Mod/Path/PathScripts/PathArray.py

```diff
@@ -155,54 +166,92 @@ def rotatePath(self, path, angle, centre):
 
         return newPath
 
-    def execute(self, obj):
-        if obj.Base:
-            if not obj.Base.isDerivedFrom("Path::Feature"):
-                return
-            if not obj.Base.Path:
-                return
-            if not obj.Base.ToolController:
-                return
+    def calculateJitter(self, obj, pos):
+        if random.randint(0,100) < obj.JitterPercent:
+            pos.x = pos.x + random.uniform(-obj.JitterMagnitude.x, obj.JitterMagnitude.y)
+            pos.y = pos.y + random.uniform(-obj.JitterMagnitude.y, obj.JitterMagnitude.y)
+            pos.z = pos.z + random.uniform(-obj.JitterMagnitude.z, obj.JitterMagnitude.z)
+        return pos
+
 
-            obj.ToolController = obj.Base.ToolController
+    def execute(self, obj):
+        if isinstance(obj.Base, list):
+            base = obj.Base
+        else:
+            base = [obj.Base]
+
+        if len(base)>0:
```

Comment:
The consequenc if len(base) == 0 isn't visible until the end.  Looking ahead, nothing happens.  So, instead you should check for that case first and  and return if it's true, raising an error of necessary.  Currently all the subsequent code is indented and dependent on that check which is confusing.